homeworks/03-data-structure/src/binary-tree/binaryTree.ts

```diff
@@ -0,0 +1,67 @@
+import { assertNever } from '../utils/commonUtils';
+
+export type TreeNode<T> = {
+  value: T;
+  left: TreeNode<T> | null;
+  right: TreeNode<T> | null;
+};
+
+export enum TraverseType {
+  depthFSInOrder = 'DFS_INORDER',
+  depthFSPreOrder = 'DFS_PREORDER',
+  depthFSPostOrder = 'DFS_POSTORDER',
+  breadthFS = 'BFS',
+}
+
+export interface Tree<T> {
+  setTreeNode(treeNode: TreeNode<T>): this;
+  traverse(type: TraverseType): T[];
+  getColumn(column: number): T[];
+}
+
+export class BinaryTree<T> implements Tree<T> {
+  protected rootNode: TreeNode<T>;
+
+  constructor(treeNode: TreeNode<T>) {
+    this.rootNode = treeNode;
+  }
```

Comment:
Just as another option: `constructor(protected treeNode: TreeNode<T>) {}`

Reply:
Updated, together with moving rules `no-useless-constructor`, `no-empty-function` to the `@typescript-eslint` - https://github.com/vbarbasins/evo-ts-bootcamp/pull/3/commits/12f47602158748ca5ea3d310f5f99edff9dcf589